### PR TITLE
feat: Pandas and GeoPandas samples

### LINKED ISSUE #231: Create pandas samples
Create samples showing:

- Reading data into pandas using `read_sql`.
- Reading data into geopandas.

### PR DESCRIPTION
**Pull request by jimfulton**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-sqlalchemy/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #231 🦕

### CODE REVIEW

**Comment by jimfulton** [comment]
I added a "do not merge" tag, because I'm not sure we want people to do this.

However, I'd kind of like to know how to do samples, and would like some feedback. :)

File: samples/snippets/pandas_read_sql.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+
+def read_data_into_pandas_using_read_sql() -> None:
+    # [START sqlalchemy_bigquery_read_sql]
```
Comment (tswast) [inline samples/snippets/pandas_read_sql.py:23]: Needs to start with `bigquery`, how about `bigquery_pandas_sqlalchemy`?

**Review by tswast — COMMENTED** [review]
These look good! I think we just need to adjust the tags so that the snippet bot is happy.

File: samples/snippets/geopandas_read_sql.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                  ST_ASBINARY(ST_GEOGPOINT(longitude, latitude)) as location
+                  from bigquery-public-data.austin_311.311_service_requests
+           limit 10
+        """,
+        engine,
+        "location",
+        "EPSG:4326",
+    )
+
+    # Don't wrap pr elide:
```
[the remark this answers is not in the document]
Comment (tseaver) [reply samples/snippets/geopandas_read_sql.py:69]: I'm betting it is really `or elide:`.

File: samples/snippets/pandas_read_sql.py (hunk 1 above)
Comment (jimfulton) [reply samples/snippets/pandas_read_sql.py]: fixed

File: samples/snippets/geopandas_read_sql.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jimfulton) [reply samples/snippets/geopandas_read_sql.py]: Yup, fixed

File: samples/snippets/geopandas_read_sql.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+
+def read_geographic_data_into_pandas_using_read_sql() -> None:
+    # [START sqlalchemy_bigquery_read_postgis]
```
[the remark this answers is not in the document]
Comment (jimfulton) [reply samples/snippets/geopandas_read_sql.py]: fixed

**Comment by tswast** [comment]
Kokoro failure looks like a flake by the backend. I learned that the backend can take as long as 4 minutes to fail on some API requests, so hopefully https://github.com/googleapis/python-bigquery/pull/859 helps with this.

```
google.api_core.exceptions.RetryError: Deadline of 120.0s exceeded while calling functools.partial(functools.partial(<bound method JSONConnection.api_request of <google.cloud.bigquery._http.Connection object at 0x7fe2a9e0e970>>, method='GET', path='/projects/precise-truck-742/queries/0aba5d5e-abdb-415a-9ffd-ad2aa450ad59', query_params={'maxResults': 0, 'location': 'US'}, timeout=None)), last exception: 500 GET https://bigquery.googleapis.com/bigquery/v2/projects/precise-truck-742/queries/0aba5d5e-abdb-415a-9ffd-ad2aa450ad59?maxResults=0&location=US&prettyPrint=false: An internal error occurred and the request could not be completed.
```

In the meantime, I'll manually trigger the Kokoro job to rebuild.

**Comment by jimfulton** [comment]
> Kokoro failure looks like a flake by the backend. 

Yup, I figured.

...

> In the meantime, I'll manually trigger the Kokoro job to rebuild.

Cool, but remember the goal of this PR, at least for me, was to get a grip on samples. :)

It's been super helpful, but I don't intend to merge it.

IMO, these samples are counter productive, because, as you've noted, the BQ python libary's `to_dataset` (and soon, `to_geodataset`) is more performant and scalable.  Of course, if you disagree, I'm happy to defer.

My thought is when `to_geodataset` lands (https://github.com/googleapis/python-bigquery/pull/848 is ready for review, BTW :) ), I'd make a GeoPandas PR to add `from_bigquery` and `to_bigquery` using `to_geodataframe` and `load_from_dataframe`.

**Comment by jimfulton** [comment]
I've merged this into the geography branch and deleted these two examples (and added a geography sample).

If you really want _these_ samples, let me know and I'll reopen this.

**Pull request closed without merging**